WalletSdk/src/main/java/com/spruceid/wallet/sdk/KeyManager.kt

```diff
@@ -0,0 +1,315 @@
+     */
+    fun reset() {
+        val ks = getKeyStore()
+        ks.aliases().iterator().forEach {
+            ks.deleteEntry(it)
+        }
+    }
+
+    /**
+     * Generates a secp256r1 signing key by id/alias in the Keystore with Strongbox when
+     * min SDK and hardware requirements are met, otherwise using TEE.
+     * @property id of the secret key.
+     * @returns KeyManagerEnvironment indicating the environment used to generate the key.
+     */
+    fun generateSigningKey(id: String): KeyManagerEnvironment {
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                generateSigningKeyWithStrongbox(id)
+
+                return KeyManagerEnvironment.Strongbox
+            } else {
+                generateSigningKeyTEE(id)
+
+                return KeyManagerEnvironment.TEE
+            }
+        } catch (e: Exception) {
+            generateSigningKeyTEE(id)
+
+            return KeyManagerEnvironment.TEE
+        }
+    }
+
+    /**
+     * Generates a secp256r1 signing key by id/alias in the Keystore with Strongbox.
+     * @property id of the secret key.
+     */
+    @RequiresApi(Build.VERSION_CODES.P)
+    private fun generateSigningKeyWithStrongbox(id: String) {
+        val generator = KeyPairGenerator.getInstance(
+            KeyProperties.KEY_ALGORITHM_EC,
+            "AndroidKeyStore",
+        )
+    
+        val spec = KeyGenParameterSpec.Builder(
+            id,
+            KeyProperties.PURPOSE_SIGN
+                    or KeyProperties.PURPOSE_VERIFY,
+        )
+            .setDigests(KeyProperties.DIGEST_SHA256, KeyProperties.DIGEST_SHA512)
+            .setAlgorithmParameterSpec(ECGenParameterSpec("secp256r1"))
+            .setIsStrongBoxBacked(true)
+            .build()
+    
+        generator.initialize(spec)
+        generator.generateKeyPair()
+    }
+
+    /**
+     * Generates a secp256r1 signing key by id/alias in the Keystore with TEE.
+     * @property id of the secret key.
+     */
+    private fun generateSigningKeyTEE(id: String) {
+        val generator = KeyPairGenerator.getInstance(
+            KeyProperties.KEY_ALGORITHM_EC,
+            "AndroidKeyStore",
+        )
+
+        val spec = KeyGenParameterSpec.Builder(
+            id,
+            KeyProperties.PURPOSE_SIGN
+                    or KeyProperties.PURPOSE_VERIFY,
+        )
+            .setDigests(KeyProperties.DIGEST_SHA256, KeyProperties.DIGEST_SHA512)
+            .setAlgorithmParameterSpec(ECGenParameterSpec("secp256r1"))
+            .build()
+
+        generator.initialize(spec)
+        generator.generateKeyPair()
+    }
+
+    /**
+     * Assumes the value above 32 will always be 33.
+     * BigInteger will add an extra byte to keep the number positive.
+     * But the key values will always be 32 bytes.
+     * @property input byte array to be processed.
+     * @return byte array with 32 bytes.
+     */
+    fun clampOrFill(input: ByteArray): ByteArray {
```

Comment:
I'm not fond of the "ByteArray! is deprecated" warnings I see, can we change to whatever the new non-deprecated type is?

Reply:
Ah, yes.

Reply:
Talked about this as a team and this should be ok.